src/Compilers/CSharp/Test/Symbol/Compilation/GetImportScopesTests.cs

```diff
@@ -0,0 +1,753 @@
+using S = System;
+/*pos*/
+class C
+{
+}";
+            var tree = Parse(text);
+            var comp = CreateCompilation(tree);
+            var model = comp.GetSemanticModel(tree);
+            var scopes = model.GetImportScopes(GetPositionForBinding(text));
+            Assert.Empty(scopes);
+        }
+
+        [Fact]
+        public void TestBeforeAliasTopLevelStatements()
+        {
+            var text = @"
+/*pos*/
+using S = System;
+
+return;";
+            var tree = Parse(text);
+            var comp = CreateCompilation(tree);
+            var model = comp.GetSemanticModel(tree);
+            var scopes = model.GetImportScopes(GetPositionForBinding(text));
+            Assert.Single(scopes);
+            Assert.Single(scopes.Single().Aliases);
+            Assert.True(scopes.Single().Aliases.Single() is { Name: "S", Target: INamespaceSymbol { ContainingNamespace.IsGlobalNamespace: true, Name: nameof(System) } });
+            Assert.True(scopes.Single().Aliases.Single().DeclaringSyntaxReferences.Single().GetSyntax() is UsingDirectiveSyntax);
+            Assert.Empty(scopes.Single().Imports);
+            Assert.Empty(scopes.Single().ExternAliases);
+            Assert.Empty(scopes.Single().XmlNamespaces);
+        }
+
+        [Fact]
+        public void TestAfterAliasTopLevelStatements1()
+        {
+            var text = @"
+using S = System;
+/*pos*/
+return;";
+            var tree = Parse(text);
+            var comp = CreateCompilation(tree);
+            var model = comp.GetSemanticModel(tree);
+            var scopes = model.GetImportScopes(GetPositionForBinding(text));
+            Assert.Empty(scopes);
+        }
+
+        [Fact]
+        public void TestAfterAliasTopLevelStatements2()
+        {
+            var text = @"
+using S = System;
+
+return /*pos*/;";
+            var tree = Parse(text);
+            var comp = CreateCompilation(tree);
+            var model = comp.GetSemanticModel(tree);
+            var scopes = model.GetImportScopes(GetPositionForBinding(text));
+            Assert.Single(scopes);
+            Assert.Single(scopes.Single().Aliases);
+            Assert.True(scopes.Single().Aliases.Single() is { Name: "S", Target: { ContainingNamespace.IsGlobalNamespace: true, Name: nameof(System) } });
+            Assert.True(scopes.Single().Aliases.Single().DeclaringSyntaxReferences.Single().GetSyntax() is UsingDirectiveSyntax);
+            Assert.Empty(scopes.Single().Imports);
+            Assert.Empty(scopes.Single().ExternAliases);
+            Assert.Empty(scopes.Single().XmlNamespaces);
+        }
+
+        [Fact]
+        public void TestAfterMultipleAliasesNoContent()
+        {
+            var text = @"
+using S = System;
+using M = Microsoft;
+/*pos*/";
+            var tree = Parse(text);
+            var comp = CreateCompilation(tree);
+            var model = comp.GetSemanticModel(tree);
+            var scopes = model.GetImportScopes(GetPositionForBinding(text));
```

Comment:
nit: all the tests start with the same steps to build `scopes`. Consider extracting.

Reply:
sure!